packages/eslint-plugin/src/rules/no-array-constructor.ts

```diff
@@ -43,7 +42,10 @@ export default createRule({
               return fixer.replaceText(node, '[]');
             }
             const fullText = context.sourceCode.getText(node);
-            const preambleLength = node.callee.range[1] - node.range[0];
+            const preambleLength =
+              node.parent.type === AST_NODE_TYPES.ChainExpression
+                ? node.callee.range[1] + 2 - node.range[0]
```

Comment:
> ` + 2`

[Testing] 😬 Hardcoded lengths like `'?'.length` & equivalent are a bit dangerous to rely on. They tend to be broken over time by AST oddities like comments and whitespace. Since this rule logic is newly dependent on character lengths, could you please add some tests to make sure this is working as intended in those edge cases?

Here's a "fun" one: 

```ts
/* a */ /* b */ Array /* c */ /* d */ ?. /* e */ /* f */ () /* g */ /* h */
```

Reply:
Added test code by referencing upstream and improved it to avoid hardcoding.